src/csm_testing/tests/sat_functional/__main__.py

```diff
@@ -43,12 +43,96 @@
   `sat_functional version.TestVersion.test_version`
 """
 import argparse
+import json
+import os
 import sys
+from typing import List
 import unittest
-import os
 
+from yaml import safe_dump
+
+TEST_PACKAGE = 'csm_testing.tests.sat_functional'
+
+
+def get_test_case_names() -> List[str]:
+    """Get a list of all the unittest.TestCase classes in the sat_functional package.
+
+    Returns:
+        The sorted list of the unittest.TestCase class names, relative to the TEST_PACKAGE.
+
+    Raises:
+        RuntimeError: If a discovered test case does not start with the TEST_PACKAGE name.
+    """
+    def get_test_cases_from_suite(suite: unittest.TestSuite) -> List[str]:
+        """Recursively get all test ids from a TestSuite"""
+        test_cases = []
+        for item in suite:
+            if isinstance(item, unittest.TestSuite):
+                test_cases.extend(get_test_cases_from_suite(item))
+            elif isinstance(item, unittest.TestCase):
+                test_cases.append(item.id())
+        return test_cases
+
+    loader = unittest.TestLoader()
+    main_suite = loader.discover(TEST_PACKAGE)
+    full_test_names = get_test_cases_from_suite(main_suite)
+
+    rel_test_names = set()
+    for test_name in full_test_names:
+        if test_name.startswith(TEST_PACKAGE + '.'):
+            # Strip off the TEST_PACKAGE name from the beginning and the test method name from the end
+            rel_test_names.add(test_name[len(TEST_PACKAGE) + 1:].rsplit('.', maxsplit=1)[0])
+        else:
+            raise RuntimeError(f'Discovered test case "{test_name}" does not start with "{TEST_PACKAGE}"')
+
+    return sorted(list(rel_test_names))
+
+
+def list_tests(parsed_args: argparse.Namespace) -> None:
+    """Print the names of all available test cases.
+
+    Args:
+        parsed_args (argparse.Namespace): The parsed arguments from the command line.
+    """
+    test_case_names = get_test_case_names()
+    if parsed_args.format == 'json':
+        print(json.dumps({'sat_tests': test_case_names}))
+    if parsed_args.format == 'yaml':
```

Comment:
Oops, this should be an `elif`.

```suggestion
    elif parsed_args.format == 'yaml':
```

This will make life hard for @ethanholen-hpe.